roachtest: update ORMs under test

rafiss opened the pull request:
fixes https://github.com/cockroachdb/cockroach/issues/79208

- typeorm: also fixes permission issues
- sqlalchemy
- pgx
- lib/pq
- pgjdbc
  -  also use secure mode for pgjdbc test. this lets us test SCRAM functionality.
  - includes a small error code change: pgwire: use InvalidPassword error code when appropriate

Release note: None

rafiss committed:
roachtest: update ORMs under test

- typeorm
- sqlalchemy
- pgjdbc (required updating expected failures)
- pgx

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79257)
<!-- Reviewable:end -->

rafiss commented on pkg/cmd/roachtest/tests/typeorm.go:
```diff
@@ -24,7 +24,7 @@ import (
 
 var typeORMReleaseTagRegex = regexp.MustCompile(`^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<point>\d+)$`)
-var supportedTypeORMRelease = "0.2.32"
+var supportedTypeORMRelease = "0.3.4"
```
holding off until 0.3.5 is released since we want https://github.com/typeorm/typeorm/commit/ec6c11920b70132396cd2e098b378b753f6e4d3f

rafiss committed:
roachtest: use secure mode for pgjdbc test

This lets us test SCRAM functionality.

Additionally, update the config so more tests pass.

Release note: None

rafiss committed:
pgwire: use InvalidPassword error code when appropriate

Release note (sql change): Use the InvalidPassword error code when
authenticating if the password is invalid or the user does not exist.

rafiss committed:
roachtest: fix permissions in typeorm testt

Release note: None

rafiss commented:
bors r-

rafiss commented:
bors r=otan

rafiss commented:
bors r=otan
